chartfx-dataset/src/test/java/de/gsi/dataset/spi/DataSetBuilderTests.java

```diff
@@ -335,23 +360,146 @@ public void testMultiDimDataSet5() {
         });
+        // request type without errors but add errors
+        assertThrows(UnsupportedOperationException.class, () -> {
+            new DataSetBuilder().setNegError(DIM_Y, new float[] { 1, 2 }).build(DefaultDataSet.class);
+        });
+        // request Editable Data Set
+        assertThrows(UnsupportedOperationException.class, () -> {
+            new DataSetBuilder() //
+                    .setValues(DIM_Y, new float[] { 1, 2 }) //
+                    .build(DoubleErrorDataSet.class);
+        });
+    }
+
+    @Test
+    public void testHelperFunctions() {
+        assertDoesNotThrow(() -> {
+            final DataSetBuilder builder = new DataSetBuilder("foo");
+            DataSet test0 = new MinimalDataSet();
+            builder.addMetaData(test0);
+            builder.setMetaErrorList("foo").setMetaInfoList("asdf", "qwert").setMetaWarningList("123", "321").setMetaInfoMap(Map.of("key", "value"));
+            DoubleErrorDataSet test1 = new DoubleErrorDataSet("test");
+            builder.addMetaData(test1);
+            builder.addMetaData(test0);
+        });
+        assertDoesNotThrow(() -> {
+            final DataSetBuilder builder = new DataSetBuilder("foo");
+            DataSet test0 = new MinimalDataSet();
+            builder.addDataLabelStyleMap(test0);
+            builder.setDataLabelMap(Map.of(1, "test", 2, "label")).setDataStyleMap(Map.of(1, "color:red"));
+            DoubleErrorDataSet test1 = new DoubleErrorDataSet("test", 2);
+            builder.addDataLabelStyleMap(test1);
+            builder.addDataLabelStyleMap(test0);
+        });
+    }
+
+    private static class MinimalDataSet implements DataSet {
+        private static final long serialVersionUID = 1L;
+        AtomicBoolean autoNotify = new AtomicBoolean();
+
+        @Override
+        public AtomicBoolean autoNotification() {
+            return autoNotify;
+        }
+
+        @Override
+        public List<EventListener> updateEventListener() {
+            return Collections.emptyList();
+        }
+
+        @Override
+        public double get(int dimIndex, int index) {
+            return dimIndex * 10 + index;
+        }
+
+        @Override
+        public List<AxisDescription> getAxisDescriptions() {
+            return List.of(new DefaultAxisDescription(this, "x Axis", "", 0.0, 9.0),
+                    new DefaultAxisDescription(this, "y Axis", "", 0.0, 90.0));
+        }
+
+        @Override
+        public int getDataCount(int dimIndex) {
+            return 10;
+        }
+
+        @Override
+        public String getDataLabel(int index) {
+            return null;
+        }
+
+        @Override
+        public int getDimension() {
+            // TODO Auto-generated method stub
+            return 0;
+        }
+
+        @Override
+        public int getIndex(int dimIndex, double value) {
+            return dimIndex == DataSet.DIM_X ? (int) Math.max(0.0, Math.min(9.0, Math.round(value))) : 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'DataSet.DIM_X' due to existing static import 'de.gsi.dataset.DataSet.DIM_X'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5594966)